src/term-search.ts

```diff
@@ -86,4 +88,25 @@ export class TermSearch {
         return new Occurrence(term, foundPositions);
     }
 
+    public static searchTermsInStemMap(stems : { [stem : string] : Array<WebPage> }) : Array<WebPage> {
+        let positives = new Set();
+        let terms = TermSearchLoader.load('terms/dbpedia_Classical_musicians_by_instruments_and_nationality.json');
+        let stemCount = Object.keys(stems).length;
+
+        //Get each webpage that is matching a search term.
+        //O(n^3) but shouldn't matter much since this is only done once per file
+        for(let i = 0; i < terms.length; i++) {
+            for(let key in stems) {
+                if(terms[i] == key) {
```

Comment:
If I understand correctly you're iterating over all keys (stems) in the associative array? Associative arrays are hash maps. You can just do `if (stems[terms[i]] !== undefined)` instead which should reduce the complexity to O(n) instead of O(n^2).